recipes/cpp-httplib/all/conanfile.py

```diff
@@ -9,20 +9,41 @@ class CpphttplibConan(ConanFile):
     topics = ("conan", "cpp-httplib", "http", "https", "header-only")
     homepage = "https://github.com/yhirose/cpp-httplib"
     url = "https://github.com/conan-io/conan-center-index"
-    settings = "os"
-    options = {"with_openssl": [True, False], "with_zlib": [True, False]}
-    default_options = {"with_openssl": False, "with_zlib": False}
     no_copy_source = True
+    settings = "os", "compiler"
+    options = {
+        "with_openssl": [True, False],
+        "with_zlib": [True, False],
+        "with_brotli": [True, False]
+    }
+    default_options = {
+        "with_openssl": False,
+        "with_zlib": False,
+        "with_brotli": False
+    }
 
     @property
     def _source_subfolder(self):
         return "source_subfolder"
 
+    def config_options(self):
+        if tools.Version(self.version) < "0.7.2":
+            del self.options.with_brotli
```

Comment:
Maybe raise ConanInvalidConfiguration instead to let user know that they are not going to get brotli despite writing with_brotli=True in their conanfile?

Reply:
They have no reason to write with_brotli=True, because conan inspect won't even display this option I guess.
I think also that deleting options based on version makes recipes more consistent. For example, recipes split in different folders don't introduce logic based on completly different versions.

For example, if you have a recipe split in 5.x.x and 6.x.x. Do you expect 5.x.x conanfile having logic based on 6.x.x?

Reply:
I understand your reasoning. Thank you for clarification.